modules/Setup/README.md

```diff
@@ -0,0 +1,208 @@
+
+with the following payload format:
+```json
+[
+    {
+        "bssid": "00:11:22:33:44:55",
+        "flags": [
+            "WPA2-PSK-CCMP",
+            "ESS"
+        ],
+        "frequency": 2437,
+        "signal_level": -41,
+        "ssid": "Example"
+    },
+    {
+        "bssid": "66:77:88:99:aa:bb",
+        "flags": [
+            "WPA2-PSK-CCMP",
+            "ESS"
+        ],
+        "frequency": 5180,
+        "signal_level": -56,
+        "ssid": "Example2"
+    }
+]
+```
+additionally general network device information is published on the following topic:
+__everest_api/setup/var/network_device_info__
+
+with the following payload format:
+```json
+[
+    {
+        "blocked": false,
+        "interface": "wlan0",
+        "ipv4": "192.0.2.23",
+        "ipv6": "",
+        "rfkill_id": "0",
+        "wireless": true
+    },
+    {
+        "blocked": false,
+        "interface": "eth0",
+        "ipv4": "192.0.2.42",
+        "ipv6": "",
+        "rfkill_id": "",
+        "wireless": false
+    }
+]
+```
+
+additionally the list of configured wifi networks is published to the following topic:
+__everest_api/setup/var/configured_networks__
+
+with the following payload format:
+```json
+[
+    {
+        "connected": true,
+        "interface": "wlan0",
+        "network_id": 0,
+        "ssid": "Example"
+    },
+    {
+        "connected": false,
+        "interface": "wlan0",
+        "network_id": 1,
+        "ssid": "Example2"
+    }
+]
+```
+
+### everest_api/setup/cmd/enable_wifi_scanning
+If any arbitrary payload is published to this topic the list of available wifi networks and general network device information just mentioned is published periodically.
+
+### everest_api/setup/cmd/disable_wifi_scanning
+If any arbitrary payload is published to this topic the list of available wifi networks and general network device information stops being periodically published.
+
+### everest_api/setup/cmd/rfkill_unblock
+If a rfkill_id is published to this topic the wifi interface with this id will be unblocked.
+
+### everest_api/setup/cmd/rfkill_block
+If a rfkill_id is published to this topic the wifi interface with this id will be blocked.
+
+### everest_api/setup/cmd/list_configured_networks
+If any arbitrary payload is published to this topic the list of configured wifi networks is published to the following topic:
+__everest_api/setup/var/configured_networks__
+
+with the following payload format:
+```json
+[
+    {
+        "connected": true,
+        "interface": "wlan0",
+        "network_id": 0,
+        "ssid": "Example"
+    },
+    {
+        "connected": false,
+        "interface": "wlan0",
+        "network_id": 1,
+        "ssid": "Example2"
+    }
+]
+```
+
+### everest_api/setup/cmd/add_network
+To add a wifi network a payload with the following format must be published to this topic:
+```json
+{
+    "interface": "wlan0",
+    "ssid": "Example",
+    "psk": "a_valid_pre_shared_key"
```

Comment:
I wonder if we should put the psk unencrypted onto mqtt

Reply:
Yeah, it would be great if we had a different way of doing this. But at the moment this mqtt connection is the only data link we have to the application